### ISSUE TITLE
OAS reconstruction results in unusual endpoints

### ISSUE THREAD

**Issue by sedkis** [apiclarity/apiclarity#73]
I spun up Tyk / APIClarity and used the good ol' `httpbin` endpoint.

I make the following API requests:
```
$ curl -s localhost:8080/httpbin/get &
    curl -s localhost:8080/httpbin/post -X POST &
    curl -s localhost:8080/httpbin/xml &
    curl -s localhost:8080/httpbin/json &
    curl -s localhost:8080/httpbin/headers &
    curl -s localhost:8080/httpbin/delete -X DELETE &
```

Two weird things happen

1) APIClarity never picks up the `xml` endpoint
2) The reconstructured OAS spec flow doesn't make sense.

Before I continue for the second one, you can recreate this by following [this branch](https://github.com/TykTechnologies/tyk-oss-k8s-deployment/tree/apiclarity).

OK so once I generate load against httpbin, I attempt to reconstruct the OAS spec.

![ezgif-2-fa36cfb4fa43](https://user-images.githubusercontent.com/8964621/146086622-858316b8-b88f-4bc4-b3b9-34c5b7cdff1e.gif)

Again (note that `xml` is missing from list of endpoints.

Number two, I can't edit the documentation at all.  And once it's generated, it's basically an empty OAS spec.
<img width="1680" alt="image" src="https://user-images.githubusercontent.com/8964621/146086747-1a4892fc-93ed-44b3-b2c8-4f7a788dab58.png">

Am I misunderstanding how to use the software?

**Comment by akpsgit** [comment]
Hi @sedkis, Thanks a lot for the detailed information.
Regarding (1), the xml is probably detected as a non-API event. Can you please go the the API events screen an turn the "show non-API" filter on and see if you get the call:
![image](https://user-images.githubusercontent.com/39153880/146190843-e63a6855-082d-4ce2-8240-cce8bb8ed078.png)

We are are reproducing (2) and looking into it, can you please maybe check if there are any errors in the apiclarity pod logs, and attach them if such exist?

Again, thanks a lot for bringing this into our attention.

**Comment by fishkerez** [comment]
A fix has been pushed to the problem.
The issue was that the leading slash of the path was sometimes removed, if the api listen path was defined with a trailing slash, and remove listen path is true.
So if the listen path in tyk api definition was "/some-path/", and the request was <address>/some-path/real-path, because the listen path is being remove, the actual path we get in the plugin is just real-path (instead of /real-path).
The simple solution was to add a leading slash if one does not exist in the plugin.

Again, thanks a lot for your feedback @sedkis. much appreciated.

**Comment by sedkis** [comment]
> 

I don't see the "Show non-API events" toggle in my version `v0.6.1` of APIClarity which I've deployed using helm

**Comment by FrimIdan** [comment]
> I don't see the "Show non-API events" toggle in my version v0.6.1 of APIClarity which I've deployed using helm

Can you please try and click on `Filters`

**Comment by sedkis** [comment]
> > I don't see the "Show non-API events" toggle in my version v0.6.1 of APIClarity which I've deployed using helm
> 
> Can you please try and click on `Filters`

I see it now, unfortunately `xml` did not show up, but that's already been adressed in #74 

Now, just waiting on a response from your team for (2), the empty OAS spec being generated.
Let me know if any additional info needed from my side.

**Comment by FrimIdan** [comment]
@sedkis #74 should fix also (2)

**Comment by sedkis** [comment]
Ahhh, my mistake I misunderstood.
Excellent!  Will be on the look out for the new Helm release.

**Comment by FrimIdan** [comment]
@sedkis we just released [v0.7.0](https://github.com/apiclarity/apiclarity/releases/tag/v0.7.0) 
Thanks again for you feedback!
